src/browser/replay/recorder.js

```diff
@@ -179,20 +179,34 @@ export default class Recorder {
     this._isReady = false;
   }
 
-  _collectEvents() {
-    const events = this._events.previous.concat(this._events.current);
+  _collectEvents(afterCount = 0) {
+    const allEvents = this._events.previous.concat(this._events.current);
 
-    // Helps the application correctly align playback by adding a noop event
-    // to the end of the recording.
-    events.push({
-      timestamp: Date.now(),
-      type: EventType.Custom,
-      data: { tag: 'replay.end', payload: {} },
-    });
+    const events = afterCount > 0 ? allEvents.slice(afterCount) : allEvents;
+
+    if (events.length > 0) {
+      // Helps the application correctly align playback by adding a noop event
+      // to the end of the recording.
+      events.push({
+        timestamp: Date.now(),
+        type: EventType.Custom,
+        data: { tag: 'replay.end', payload: {} },
+      });
+    }
```

Comment:
Right now we'll be sending two `replay.end` events. One for the pre span and another for the post span. I'm not sure if this is what we want.